x/incentive/keeper/grpc_query.go

```diff
@@ -0,0 +1,331 @@
+package keeper
+
+import (
+	"context"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/kava-labs/kava/x/incentive/types"
+	liquidtypes "github.com/kava-labs/kava/x/liquid/types"
+)
+
+type queryServer struct {
+	keeper Keeper
+}
+
+var _ types.QueryServer = queryServer{}
+
+// NewQueryServerImpl creates a new server for handling gRPC queries.
+func NewQueryServerImpl(keeper Keeper) types.QueryServer {
+	return &queryServer{
+		keeper: keeper,
+	}
+}
+
+func (s queryServer) Params(
+	ctx context.Context,
+	req *types.QueryParamsRequest,
+) (*types.QueryParamsResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	return &types.QueryParamsResponse{
+		Params: s.keeper.GetParams(sdkCtx),
+	}, nil
+}
+
+func (s queryServer) Rewards(
+	ctx context.Context,
+	req *types.QueryRewardsRequest,
+) (*types.QueryRewardsResponse, error) {
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	res := types.QueryRewardsResponse{}
+
+	var owner sdk.AccAddress
+	hasOwner := req.Owner != ""
+	if req.Owner != "" {
+		addr, err := sdk.AccAddressFromBech32(req.Owner)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "invalid address: %s", err)
+		}
+
+		owner = addr
+	}
+
+	if err := s.queryRewards(sdkCtx, &res, owner, hasOwner, req.RewardType); err != nil {
+		return nil, err
+	}
+
+	if !req.Unsynchronized {
+		if err := s.synchronizeRewards(sdkCtx, &res); err != nil {
```

Comment:
curious if this query request result in state changes?

Reply:
on further reading looks like this method only results in simulated state changes 😅